x-pack/test/functional/apps/dev_tools/searchprofiler_editor.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { KibanaFunctionalTestDefaultProviders } from '../../../types/providers';
+
+// eslint-disable-next-line import/no-default-export
+export default function({ getPageObjects, getService }: KibanaFunctionalTestDefaultProviders) {
+  const PageObjects = getPageObjects(['common']);
+  const testSubjects = getService('testSubjects');
+  const aceEditor = getService('aceEditor');
+  const retry = getService('retry');
+
+  const editorTestSubjectSelector = 'searchProfilerEditor';
+
+  describe('Search Profiler Editor', () => {
+    before(async () => {
+      await PageObjects.common.navigateToApp('searchProfiler');
+      expect(await testSubjects.exists('searchProfilerEditor')).to.be(true);
+    });
+
+    it('correctly parses triple quotes in JSON', async () => {
+      // The below inputs are written to work _with_ ace's autocomplete unlike console's unit test
+      // counterparts in src/legacy/core_plugins/console/public/tests/src/editor.test.js
+
+      const okInput = [
+        `{
+"query": {
+"match_all": {}`,
+        `{
+"query": {
+"match_all": {
+"test": """{ "more": "json" }"""`,
+      ];
+
+      const nOkInput = [
+        `{
+"query": {
+"match_all": {
+"test": """{ "more": "json" }""`,
+        `{
+"query": {
+"match_all": {
+"test": """{ "more": "json" }""'`,
+      ];
+
+      const expectHasParseErrorsToBe = (expectation: boolean) => async (inputs: string[]) => {
```

Comment:
For sure, I was kinda stuffing around a bit here :p. Very happy to make changes - I just liked the way it read: `expectHasParseErrorsToBe(false)(okInput)`. But that's really subjective :).

Reply:
Personally, I think I'd just prefer to see await expectHasParseErrorsToBe(false, okInput);` since the partial application doesn't seem to offer any benefits here, unless I'm missing something.

Likewise, it'd simplify the code slightly if `inputs` was just a string, since we're not defining multiple types per case in this test.

But these are just nits. :)